packages/backend/src/services/UserService.ts

```diff
@@ -236,4 +251,51 @@ export class UserService {
         });
         return user;
     }
+
+    async getPasswordResetLink(code: string): Promise<PasswordResetLink> {
+        const link = await this.passwordResetLinkModel.getByCode(code);
+        const now = new Date();
+        if (link.expiresAt <= now) {
+            try {
+                await this.passwordResetLinkModel.deleteByCode(link.code);
+            } catch (e) {
+                throw new NotExistsError('Password reset link not found');
+            }
+            throw new NotExistsError('Password reset link expired');
+        }
+        return link;
```

Comment:
Should this be `verifyPasswordResetLink` and not return info? Just to reduce the chance we expose PasswordResetLink via the API, since it contains the email